cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -4927,6 +4927,161 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                 }
             }
         }
+        Opcode::IaddPairwise => {
+            if let (Some(swiden_low), Some(swiden_high)) = (
+                matches_input(ctx, inputs[0], Opcode::SwidenLow),
+                matches_input(ctx, inputs[1], Opcode::SwidenHigh),
+            ) {
+                let swiden_input = &[
+                    InsnInput {
+                        insn: swiden_low,
+                        input: 0,
+                    },
+                    InsnInput {
+                        insn: swiden_high,
+                        input: 0,
+                    },
+                ];
+
+                let input0_ty = ctx.input_ty(swiden_low, 0);
+                let input1_ty = ctx.input_ty(swiden_high, 0);
+                let output_ty = ctx.output_ty(insn, 0);
+                let src = put_input_in_reg(ctx, swiden_input[0]);
+                let dst = get_output_reg(ctx, outputs[0]).only_reg().unwrap();
+                assert_eq!(input0_ty, input1_ty);
+                assert_eq!(swiden_input[0].input, swiden_input[1].input);
```

Comment:
I don't think this is testing what we want to test -- this is just checking that we grabbed input 0 on each of the swiden ops (which is always true by construction, since we built `swiden_input` just above).

What we really want is to `put_input_in_reg(ctx, swiden_input[1])`, and then assert that we have the same register, I think.

Also, if the inputs do *not* match, we should fail more gracefully than with an assert, I think; asserts imply a compiler bug, whereas the different-inputs case is really just unimplemented behavior, right? So perhaps we could do (after getting `src0` and `src1` regs) `if src0 != src1 { unimplemented!("iadd_pairwise not implemented for general case with different inputs"); }` or something like that?

Reply:
Sorry .. Yes. When I first did this I was not sure if that's what we wanted here and forgot to highlight that concern. Good catch. Also updated for the unimplemented vs assert comment.